dali/operators/numba_function/numba_func.cc

```diff
@@ -71,91 +66,206 @@ Also lets provide run function:
 
 .. code-block:: python
 
-    @cfunc(dali_numba.run_fn_sig(types.int32, types.int64), nopython=True)
-    def change_out_shape(out_ptr, out_shape_ptr, ndim_out, in_ptr, in_shape_ptr, ndim_in):
-        out_shape = carray(out_shape_ptr, ndim_out)
-        out_arr = carray(out_ptr, (out_shape[0], out_shape[1], out_shape[2]))
-        in_shape = carray(in_shape_ptr, ndim_in)
-        in_arr = carray(in_ptr, (in_shape[0], in_shape[1], in_shape[2]))
-        for i in range(in_shape[0]):
-            for j in range(in_shape[1]):
-                out_arr[j, i] = in_arr[i, j]
+    def run_fn(out0, in0):
+        for i in range(in0.shape[0]):
+            for j in range(in0.shape[1]):
+                out0[j, i] = in0[i, j]
 
-The run function works on a per-sample basis, so we only need to handle one sample.
-Notice that we are passing ``int32`` as the output data type, which we set in the setup function body.
+The run function works on a per-sample basis, so we only need to handle one sample. Although it's 
+possible to run it per batch (just specify ``batch_processing`` argument in operator). The run function per-batch
+may look like this:
+
+.. code-block:: python
 
+    def run_fn(out0_samples, in0_samples):
+        for sample_id in range(len(out0_samples)):
+            out0 = out0_samples[sample_id]
+            in0 = in0_samples[sample_id]
+            for i in range(in0.shape[0]):
+                for j in range(in0.shape[1]):
+                    out0[j, i] = in0[i, j]
 )code")
-  .NumInput(1)
-  .NumOutput(1)
+  .NumInput(1, 6)
+  .OutputFn([](const OpSpec &spec) { return spec.GetRepeatedArgument<int>("out_types").size(); })
+  .AllowSequences()
```

Comment:
Add `.NoPrune()`. It documents the fact that the operator will not be removed from the graph even when the outputs are unused.